src/setup/sound.c

```diff
@@ -129,10 +138,58 @@ static void OpenMusicPackDir(TXT_UNCAST_ARG(widget), TXT_UNCAST_ARG(unused))
     }
 }
 
+#ifdef _WIN32
+static void UpdateMidiDevice(TXT_UNCAST_ARG(widget), TXT_UNCAST_ARG(data))
+{
+    winmm_midi_device = midi_index - 1;
+}
+
+static txt_dropdown_list_t *MidiDeviceSelector(void)
+{
+    txt_dropdown_list_t *result;
+    UINT device_id;
+    UINT num_devices;
+    MMRESULT mmr;
+    MIDIOUTCAPS mcaps;
+    int i;
+
+    num_devices = midiOutGetNumDevs() + 1; // include MIDI_MAPPER
+    if (num_devices > 20)
+    {
+        num_devices = 20;
+    }
+    midi_names = malloc(num_devices * sizeof(char*));
```

Comment:
clang-format found a formatting issue, see the bot comment above.

Reply:
Also shouldn't this be `free`d somewhere?

Reply:
Done. This one won't be fixed: `File src/setup/sound.c (lines 301, 302, 303, 304, 305, 344): Code does not conform to Custom style guidelines.`